02-peer-protocol.md

```diff
@@ -445,6 +828,343 @@ to broadcast the commitment transaction to get his funds back and open a new
+    - MUST fail the negotiation if:
+      - the transaction's total fees is less than the last
+        successfully negotiated transaction's fees
+      - the transaction does not share a common input with all previous
+        funding transactions
+
+### The `commitment_signed` Message
+
+This message is exchanged by both peers. It contains the signatures for
+the first commitment transaction.
+
+Rationale and Requirements are the same as listed below,
+for [`commitment_signed`](#commiting-updates-so-far-commitment_signed) with the following additions.
+
+#### Requirements
+
+The sending node:
+  - MUST send zero HTLC's.
+
+The receiving node:
+  - if the message has one or more HTLC's:
+    - MUST fail the negotiation
+  - if it has not already transmitted its `commitment_signed`:
+    - MUST send `commitment_signed`
+  - Otherwise:
+    - MUST send `tx_signatures`
+
+#### Rationale
+
+The first commitment transaction has no HTLC's.
+
+
+### Sharing funding signatures: `tx_signatures`
+
+After a valid `commitment_signature` has been received
+from the peer and a `commitment_signature` has been sent, a peer:
+  - MUST transmit a [`tx_signatures` message](#the-tx_signatures-message) with their signatures for
+    the funding transaction
+
+
+#### Requirements
+The sending node:
+  - MUST verify it has received a valid commitment signature from its peer
+  - MUST remember the details of this funding transaction
+  - if it has NOT received a valid `commitment_signed` message:
+    - MUST NOT send a `tx_signatures` message
+
+The receiving node:
+  - if the received `witness_stack` weight results in the peer's
+    paid feerate falling below the *opener*'s feerate for the funding
+    transaction:
+    - SHOULD broadcast their commitment transaction, closing the channel.
+  - SHOULD apply `witness`es to the funding transaction and broadcast it
+  - if has already sent or received a `funding_locked` message for this
+    channel:
+    - MUST ignore this message
+
+#### Rationale
+A peer sends their `tx_signatures` as soon as they have received a valid
+`commitments_signed` message.
+
+The channel should be preemptively closed in the case where a peer provides
+valid witness data that causes their paid feerate to fall beneath the
+`open_channel2.funding_feerate_perkw` rate. This penalizes the peer
+for underpayment of fees.
+
+
+### The `init_rbf` Message
+
+This message initiates a replacement of a broadcast funding transaction.
+
+1. type: 72 (`init_rbf`)
+2. data:
+   * [`channel_id`:`channel_id`]
+   * [`u64`:`funding_satoshis`]
+   * [`u32`:`locktime`]
+   * [`u32`:`funding_feerate_perkw`]
+
+#### Requirements
+
+The sender:
+  - MUST have sent `open_channel2`
+  - MUST set `funding_feerate_perkw` greater than or equal to
+    65/64 times the last sent `funding_feerate_perkw` rounded down.
+  - MUST NOT have sent or received a `funding_locked` message.
+
+The recipient:
+  - MUST respond with either an error or an `ack_rbf` message.
```

Comment:
I think we shouldn't use the `error` message here, because at that point it conflicts with the requirement to abandon the channel if an `error` is received...
An explicit message to `reject_rbf` feels cleaner, as it only rejects that rbf attempt without impacting the rest of the channel flow.

Reply:
I think I mean `warning`; basically "fail the negotiation". In fact, at any point in time the peer can fail the RBF negotiation (send a warning and hang up). The peer should reconnect and re-try. Or reconnect and allow the channel open to proceed (assuming the original funding tx gets mined).